shhext/signal.go

```diff
@@ -5,17 +5,28 @@ import (
 	"github.com/status-im/status-go/geth/signal"
 )
 
-// EnvelopeSentSignal includes hash of the sent envelope.
-type EnvelopeSentSignal struct {
+// EnvelopeSignal includes hash of the envelope.
+type EnvelopeSignal struct {
 	Hash common.Hash `json:"hash"`
 }
 
-// SendEnvelopeSentSignal sends an envelope.sent signal with hash of the envelope.
-func SendEnvelopeSentSignal(hash common.Hash) {
+// EnvelopeSignalHandler sends signals when envelope is sent or expired.
+type EnvelopeSignalHandler struct{}
+
+var _ EnvelopeEventsHandler = EnvelopeSignalHandler{}
```

Comment:
If you do it like this: `var _ EnvelopeEventsHandler = (*EnvelopeSignalHandler)(nil)`, you won't allocate any memory.

Reply:
btw if the current mechanism is used than I guarantee that both reference and value satisfies the interface. i thought about it and i think i can remove this interface checking at all cause i have a test for this in the same package